### DISCUSSION TITLE
Anonymous sign-in upgrade flow

### DISCUSSION THREAD

**Discussion by Kirsdarkenvaar** [supabase/supabase#28545] [Questions]
I'm trying to add anonymous sign-in support to my Flutter mobile app. I've followed the guide in the blog post: https://supabase.com/blog/anonymous-sign-ins

Step 1: When the user starts their free trial, I call signInAnonymously(), which correctly creates a user with is_anonymous = true. So far so good.

Step 2: When the user elects to upgrade, I ask them to enter an email, then call updateUser:

`await Supabase.instance.client.auth.updateUser(
          UserAttributes(
            email: 'foo@bar.com',
          ),
          emailRedirectTo:
              kIsWeb ? null : 'com.scheme.my://email-verification/',
        );`

This has the following effects:
- Sets the "email" field of the user to "foo@bar.com".
- Sends a change confirmation email to the address.

Note: The guide says to call `updateUser({ "foo@bar.com" })` but perhaps the API has changed since the post was published?

Step 3: When the user clicks on the magic link in the email, that is picked up by the OS and opened in the app. This has the following effects:
- Sets the "email_confirmed_at" field of the user to now.
- Sets the "provider" to "email".
- Sets "is_anonymous" to false.

At this point the user appears to have been converted to a regular user.

The two main issues I'm facing with this process are:
1. How does the user set their password? Even if I ask them to provide one when they elect to upgrade in step 2, I cannot set it with updateUser().
2. How do I know that the user confirmed their email in step 3? The magic link does not trigger an auth state change.

**Comment by bcanzanella** [comment]
@Kirsdarkenvaar did you ever figure this out? I'm having a similar issue.
